# Perfis de aplicativos rentáveis para a App Store e o Google Play Markets

Nosso objetivo neste projeto é encontrar perfis de aplicativos para dispositivos móveis que sejam lucrativos para os mercados da App Store e do Google Play. Estamos trabalhando como analistas de dados para uma empresa que cria aplicativos móveis para Android e iOS, e nosso trabalho é permitir que nossa equipe de desenvolvedores tome decisões orientadas por dados com relação ao tipo de aplicativos que eles constroem.

Em nossa empresa, criamos apenas aplicativos gratuitos para baixar e instalar, e nossa principal fonte de receita consiste em anúncios no aplicativo. Isso significa que nossa receita para qualquer aplicativo é influenciada principalmente pelo número de usuários que usam nosso aplicativo. Nosso objetivo neste projeto é analisar dados para ajudar nossos desenvolvedores a entender que tipos de aplicativos provavelmente atrairão mais usuários.

In [21]:
opened_file = open('AppleStore.csv')
from csv import reader
read_file = reader(opened_file)
ios = list(read_file)
ios_header = apple_dados[0]
#apple_data = apple_data[1:]

opened_file = open('googleplaystore.csv')
read_file = reader(opened_file)
android = list(read_file)
android_header = android_dados[0]
#google_data = google_data[1:]

In [2]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

In [22]:
explore_data(ios, 0,5, True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


Number of rows: 7198
Number of columns: 16


In [23]:
explore_data(android, 0, 5, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 10842
Number of columns: 13


In [5]:
android_header

['App',
 'Category',
 'Rating',
 'Reviews',
 'Size',
 'Installs',
 'Type',
 'Price',
 'Content Rating',
 'Genres',
 'Last Updated',
 'Current Ver',
 'Android Ver']

In [24]:
ios_header

['id',
 'track_name',
 'size_bytes',
 'currency',
 'price',
 'rating_count_tot',
 'rating_count_ver',
 'user_rating',
 'user_rating_ver',
 'ver',
 'cont_rating',
 'prime_genre',
 'sup_devices.num',
 'ipadSc_urls.num',
 'lang.num',
 'vpp_lic']

In [25]:
#A linha de índice #10472 está com a categoria faltando
#Vamos remove-la
print(android_header,"\n",android[10473])
del android[10473]

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver'] 
 ['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


# Há itens duplicados no dataset Google Play  Store

Há aplicativos que possuem mais de uma entrada no dataset. E para nossa análise, é importante ter só uma instância de cada.
Observe abaixo, onde constatamos 4 instancias do app Instagram e 2 do app Facebook.

In [26]:
for app in android[1:]:
    name = app[0]
    if name == 'Instagram':
        print(app)
    if name == 'Facebook':
        print(app)

['Facebook', 'SOCIAL', '4.1', '78158306', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'August 3, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Facebook', 'SOCIAL', '4.1', '78128208', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'August 3, 2

Observe que no caso do app Instagram e Facebook, visto acima, a diferença entre as instâncias reside no número de reviews.  O que sugere que os dados foram coletados em horas diferentes.
Para corrigir esse erro, vamos remover as duplicatas. Podemos usar o critério de que quem tiver maior número de reviews permanece.

Abaixo, contabilizamos e identificamos o número de duplicados no dataset.

In [99]:
duplicados = []
unicos = []
    
for app in android[1:]:
    nome = app[0]
    
    if nome in unicos:
        duplicados.append(nome)
    else:
        unicos.append(nome)

print('Numero de apps duplicados:', len(duplicados), 
      "\n\nExemplos de apps duplicados:", duplicados[:15])  

Numero de apps duplicados: 1181 

Exemplos de apps duplicados: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


# Removendo instâncias duplicadas

A seguir, iremos remover as entradas duplicadas, mantendo somente a entrada com maior número de review.

Vamos começar preenchendo um dicionário com o nome do app e seu respectivo número de review max. 

In [29]:
reviews_max = {}

for app in android[1:]:

    nome = app[0]
    n_reviews = float(app[3])
    
    if nome in reviews_max and reviews_max[nome] < n_reviews:
        reviews_max[nome] = n_reviews
    elif nome not in reviews_max:
        reviews_max[nome] = n_reviews
        
print("Tamanho esperado: ", len(android[1:])- len(duplicados))
print("Tamanho obtido:", len(reviews_max))

Tamanho esperado:  9659
Tamanho obtido: 9659


Agora, usaremos o dicionário criado para realmente remover os duplicados de menor número de reviews.

Para isso, vamos percorrer os dados dos aplicativos do google e verificar se o número de reviews de cada app corresponde ao seu  máx numero de reviews e se essa instância já não se encontra na nova lista, se sim, a instância é adicionada a uma nova lista que corresponde aos dados já limpos, sem instâncias duplicadas.

In [33]:
android_clean = []
adicionados = []

for app in android[1:]:
    nome = app[0]
    n_reviews = float(app[3])
    
    if (n_reviews == reviews_max[nome]) and (nome not in adicionados):
        android_clean.append(app)
        adicionados.append(nome)
        
print("Tamanho esperado:", len(reviews_max),
      "\nTamanho obtido:", len(android_clean),
      "\nTamanho dos dados não limpos:", len(android[1:]))        


Tamanho esperado: 9659 
Tamanho obtido: 9659 
Tamanho dos dados não limpos: 10840


# Removendo caracteres especiais

Nosso obejto de análise são apps direcionados á uma audiência de falantes de inglês. Se analisarmos o dataset, encontraremos apps que não são direcionados a falantes de inglês:


In [39]:
print(ios[814][1])
print(ios[6732][1],"\n")
print(android_clean[4412][0])
print(android_clean[7940][0])

爱奇艺PPS -《欢乐颂2》电视剧热播
【脱出ゲーム】絶対に最後までプレイしないで 〜謎解き＆ブロックパズル〜 

中国語 AQリスニング
لعبة تقدر تربح DZ


Textos em inglês incluem o alfabeto, números(0-9) e pontuação/símbolos (.,!,?,+,*,/) *.

Por trás dos panos, cada carácter que usamos em uma string possui um representante numeral, e a tabela que usamos para detalhar isso é a conhecida como tabela * **ASCII** (American Standard Code for Information Interchange) *.

Os números que correspondem aos carctéres comumente usados em um texto escrito em inglês variam de 0 á 127. Portanto, nós precisamos verificar se cada letra das strings no dataset está dentro desse range.

Para isso, nós temos a função * **ord( )** *, que nos retorna o número correspondente de cada caractere.

A seguir iremos criar uma função que verifica se uma determinada string é um texto inglês, ou seja, se está dentro do range 0-127:

In [41]:
def is_english(string):
    for caracter in string:
        if ord(caracter) > 127:
            return False
    return True        

Agora iremos testar a função usando algumas strings:

In [46]:
strings = ["Instagram",
           '爱奇艺PPS -《欢乐颂2》电视剧热播',
           'Docs To Go™ Free Office Suite',
           'Instachat 😜']

for string in strings:
    print(is_english(string))

True
False
False
False


Como pode ser visto acima, a função etiquetou apps em inglês como não-ingleses erroneamente, pois possuem emojis.

Para corrigir isso, só iremos etiquetar apps como não ingleses, se o app tiver mais do que 3 caractéres maiores que 127.

In [52]:
def string_em_ingles (string):
   
    ''' 
    Retorna False se a string possuir
    mais que três caracteres especias. 
    Retorna True, caso contrário.
    '''
    
    i = 0
    for caracter in string:
        if ord(caracter) > 127:
            i += 1
            if i > 3:
                return False
    return True        

In [55]:
for string in strings:
    
    print(string,": ", string_em_ingles(string),"\n")

Instagram :  True 

爱奇艺PPS -《欢乐颂2》电视剧热播 :  False 

Docs To Go™ Free Office Suite :  True 

Instachat 😜 :  True 



A função criada para separar apps em inglês de não-inglês não é 100% eficiente, mas cumpre o papel da separação no nosso caso, como testao acima.

Abaixo, vamos retirar de ambos datasets, as instâncias que não são texto em inglês.

In [62]:
#Retirando instancias com cacteres especiais

#ANDROID
android_cleaner = []

for app in android_clean:
    nome = app[0]
    if string_em_ingles(nome):
        android_cleaner.append(app)

#IOS
ios_clean = []

for app in ios:
    nome = app[1]
    if string_em_ingles(nome):
        ios_clean.append(app)


In [67]:
print("N° instâncias *android*: ", len(android_cleaner),
     "\nN° instâncias *ios*: ", len(ios_clean))

N° instâncias *android*:  9614 
N° instâncias *ios*:  6184


# Selecionando apps gratuitos

Como mencionado no inicio, nós iremos analisar somente apps que são gratuitos.

Abaixo, vamos remover os apps que possuem preço igual a zero.

Para isso vamos verificar os valores na coluna *"price"*  de cada dataset.

In [103]:
#Verificando o index da coluna price

ios_price_index = ios_header.index("price")
android_price_index = android_header.index("Price")

print("Android price index: ", android_price_index,
     "\n\nIOS price index:", ios_price_index)

Android price index:  7 

IOS price index: 4


In [104]:
# Android: Index da coluna "price" é 7
# IOS: Index da coluna "price" é 4

android_free = [app for app in android_cleaner
                if app[android_price_index] == "0"]

ios_free = [app for app in ios_clean 
            if app[ios_price_index] == "0.0"]

print("N° instâncias *android-free-apps*: ", len(android_free),
     "\nN° instâncias *ios-free-apps*: ", len(ios_free))

N° instâncias *android-free-apps*:  8864 
N° instâncias *ios-free-apps*:  3222


# Overview e início da Análise

Até agora nós "limpamos" os dados:
- Removendo instâncias com erros
- Removendo instâncias duplicadas
- Removendo apps que não são em inglês
- Isolando os apps gratuitos

Como mencionado no início, estou interessada em saber quais características de app, fazem com que ele atraia maior número de usuários. 

Como nosso objetivo é adicionar app em ambas as lojas (Google e Apple), nós precisamos encontrar um perfil de app que faça sucesso em ambas lojas.
Por exemplo, um perfil que funciona para ambas pode ser um app de produtividade que faz uso da gamificação.

## Análise de gêneros mais comuns
Vamos iniciar a análise verificando os gêneros dos app e os mais comuns em cada loja. Para isso, vamos preencher uma tabela de frequência de algumas colunas. Mas primeiro, devemos identificar quais colunas podem ser útil nessa parte da análise.

In [105]:
android_header

['App',
 'Category',
 'Rating',
 'Reviews',
 'Size',
 'Installs',
 'Type',
 'Price',
 'Content Rating',
 'Genres',
 'Last Updated',
 'Current Ver',
 'Android Ver']

In [106]:
ios_header

['id',
 'track_name',
 'size_bytes',
 'currency',
 'price',
 'rating_count_tot',
 'rating_count_ver',
 'user_rating',
 'user_rating_ver',
 'ver',
 'cont_rating',
 'prime_genre',
 'sup_devices.num',
 'ipadSc_urls.num',
 'lang.num',
 'vpp_lic']

Após analisar as colunas em cada dataset, identifiquei que no dataset do google as colunas "Category" e "Genres" são as mais apropriadas par se avaliar os gêneros mais comuns.
No caso da Apple, foi a coluna "prime_genre".

***
Como alguns nomes das colunas não são auto explicativos, aqui está o link para verificar a explicação:
IOS: https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps
Android: https://www.kaggle.com/lava18/google-play-store-apps
***

Agora, criarei uma função para auxiliar na construção de tabelas de frequencia, já que precisaremos construí-las para 3 colunas.

In [155]:
def tabela_freq(dataset, indice):
    ''' 
    Retorna uma tabela de frequências dado uma lista de listas,
    e o índice da coluna que se quer a frequencia dos itens
    '''
    tabela = {}
    for row in dataset:
        tabela[row[indice]] = tabela.get(row[indice], 0) + 1
    return tabela


def display_table(dataset, indice):
    '''
    Mostra na tela a tabela de frequência da coluna índice, do 
    dataset *dataset*, de forma ordenada descendente
    '''
    table = tabela_freq(dataset, indice)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0],'->',
              round((entry[0]/len(dataset))*100, 3),"%" )

In [156]:
print("IOS -> PRIME GENRE:\n")
display_table(ios_free,
              ios_header.index("prime_genre"))                          

IOS -> PRIME GENRE:

Games : 1874 -> 58.163 %
Entertainment : 254 -> 7.883 %
Photo & Video : 160 -> 4.966 %
Education : 118 -> 3.662 %
Social Networking : 106 -> 3.29 %
Shopping : 84 -> 2.607 %
Utilities : 81 -> 2.514 %
Sports : 69 -> 2.142 %
Music : 66 -> 2.048 %
Health & Fitness : 65 -> 2.017 %
Productivity : 56 -> 1.738 %
Lifestyle : 51 -> 1.583 %
News : 43 -> 1.335 %
Travel : 40 -> 1.241 %
Finance : 36 -> 1.117 %
Weather : 28 -> 0.869 %
Food & Drink : 26 -> 0.807 %
Reference : 18 -> 0.559 %
Business : 17 -> 0.528 %
Book : 14 -> 0.435 %
Navigation : 6 -> 0.186 %
Medical : 6 -> 0.186 %
Catalogs : 4 -> 0.124 %


### Análise da coluna "prime_genre" dos apps da Apple

Como pode ser visto acima, os gêneros mais comuns são Jogos, Entretenimento e Fotos & Videos.

*** 
- Games: 58.163 %

- Entertainment: 7.883 %

- Photo & Video: 4.966 %
***
Aparentemente, a maioria dos apps (para ios em inglês e gratuítos) são para entretenimento e diversão (Games, Entertainment, Photo & Video).

Mas não podemos recomendar um perfil de app baseado somente nessa tabela, pois não podemos inferir que por haver maior número de apps do gênero x, o número de usuários desses apps também será o majoritário.

In [152]:
print("ANDROID -> GENRE:\n")
display_table(android_free,
              android_header.index("Genres"))
                                   

ANDROID -> GENRE:

Tools : 749 : 8.45 %
Entertainment : 538 : 6.069 %
Education : 474 : 5.347 %
Business : 407 : 4.592 %
Productivity : 345 : 3.892 %
Lifestyle : 345 : 3.892 %
Finance : 328 : 3.7 %
Medical : 313 : 3.531 %
Sports : 307 : 3.463 %
Personalization : 294 : 3.317 %
Communication : 287 : 3.238 %
Action : 275 : 3.102 %
Health & Fitness : 273 : 3.08 %
Photography : 261 : 2.944 %
News & Magazines : 248 : 2.798 %
Social : 236 : 2.662 %
Travel & Local : 206 : 2.324 %
Shopping : 199 : 2.245 %
Books & Reference : 190 : 2.144 %
Simulation : 181 : 2.042 %
Dating : 165 : 1.861 %
Arcade : 164 : 1.85 %
Video Players & Editors : 157 : 1.771 %
Casual : 156 : 1.76 %
Maps & Navigation : 124 : 1.399 %
Food & Drink : 110 : 1.241 %
Puzzle : 100 : 1.128 %
Racing : 88 : 0.993 %
Role Playing : 83 : 0.936 %
Libraries & Demo : 83 : 0.936 %
Auto & Vehicles : 82 : 0.925 %
Strategy : 81 : 0.914 %
House & Home : 73 : 0.824 %
Weather : 71 : 0.801 %
Events : 63 : 0.711 %
Adventure : 60 : 0.677 %
Comics : 

In [153]:
print("ANDROID -> CATEGORY:\n")
display_table(android_free,
              android_header.index("Category"))
                                   

ANDROID -> CATEGORY:

FAMILY : 1676 : 18.908 %
GAME : 862 : 9.725 %
TOOLS : 750 : 8.461 %
BUSINESS : 407 : 4.592 %
LIFESTYLE : 346 : 3.903 %
PRODUCTIVITY : 345 : 3.892 %
FINANCE : 328 : 3.7 %
MEDICAL : 313 : 3.531 %
SPORTS : 301 : 3.396 %
PERSONALIZATION : 294 : 3.317 %
COMMUNICATION : 287 : 3.238 %
HEALTH_AND_FITNESS : 273 : 3.08 %
PHOTOGRAPHY : 261 : 2.944 %
NEWS_AND_MAGAZINES : 248 : 2.798 %
SOCIAL : 236 : 2.662 %
TRAVEL_AND_LOCAL : 207 : 2.335 %
SHOPPING : 199 : 2.245 %
BOOKS_AND_REFERENCE : 190 : 2.144 %
DATING : 165 : 1.861 %
VIDEO_PLAYERS : 159 : 1.794 %
MAPS_AND_NAVIGATION : 124 : 1.399 %
FOOD_AND_DRINK : 110 : 1.241 %
EDUCATION : 103 : 1.162 %
ENTERTAINMENT : 85 : 0.959 %
LIBRARIES_AND_DEMO : 83 : 0.936 %
AUTO_AND_VEHICLES : 82 : 0.925 %
HOUSE_AND_HOME : 73 : 0.824 %
WEATHER : 71 : 0.801 %
EVENTS : 63 : 0.711 %
PARENTING : 58 : 0.654 %
ART_AND_DESIGN : 57 : 0.643 %
COMICS : 55 : 0.62 %
BEAUTY : 53 : 0.598 %


### Análise das colunas "Genres" e "Category" dos apps do Google

As colunas "Genres" e "Category", são bem semelhantes entre si. Mas parece que a coluna "Category" contém vários generos da coluna "Genres" e muito mais. Por isso, vamos analisar ela por enquanto.

Os gêneros mais comuns são:
***
- FAMILY: 18.908 %

- GAME: 9.725 %

- TOOLS: 8.461 %
***

Pelo mesmo motivo da análise anterior, não podemos recomendar um perfil com base nisso.

# Analisando o número de downloads por gênero

Aqui nós vamos calcular a média de downloads dos apps, considerando separadamente cada gênero. 

Vamos iniciar criando uma tabela de frequência para os gêneros

## Analisando IOS apps

In [163]:
prime_genre = tabela_freq(ios_free,
              ios_header.index("prime_genre"))
dic = {}
for genre in prime_genre:
    total = 0
    len_genre = 0
    
    for app in ios_free:
        genre_app = app[ios_header.index("prime_genre")]
        
        if genre_app == genre:
            rating = float(app[ios_header.index('rating_count_tot')])
            total += rating
            len_genre += 1
            
    rating_avg = total/len_genre
    dic[genre] = rating_avg
    print("Gênero: ", genre, "| Média: ", rating_avg )

Gênero:  Social Networking | Média:  71548.34905660378
Gênero:  Lifestyle | Média:  16485.764705882353
Gênero:  Shopping | Média:  26919.690476190477
Gênero:  Catalogs | Média:  4004.0
Gênero:  Productivity | Média:  21028.410714285714
Gênero:  Music | Média:  57326.530303030304
Gênero:  News | Média:  21248.023255813954
Gênero:  Education | Média:  7003.983050847458
Gênero:  Medical | Média:  612.0
Gênero:  Photo & Video | Média:  28441.54375
Gênero:  Reference | Média:  74942.11111111111
Gênero:  Utilities | Média:  18684.456790123455
Gênero:  Book | Média:  39758.5
Gênero:  Games | Média:  22788.6696905016
Gênero:  Weather | Média:  52279.892857142855
Gênero:  Sports | Média:  23008.898550724636
Gênero:  Finance | Média:  31467.944444444445
Gênero:  Entertainment | Média:  14029.830708661417
Gênero:  Business | Média:  7491.117647058823
Gênero:  Travel | Média:  28243.8
Gênero:  Food & Drink | Média:  33333.92307692308
Gênero:  Navigation | Média:  86090.33333333333
Gênero:  Health 

In [170]:
table_display=[]
for key in dic:
    key_val_as_tuple = (dic[key], key)
    table_display.append(key_val_as_tuple)

table_sorted = sorted(table_display, reverse = True)
for entry in table_sorted:
    print(entry[1], ':', entry[0])

Navigation : 86090.33333333333
Reference : 74942.11111111111
Social Networking : 71548.34905660378
Music : 57326.530303030304
Weather : 52279.892857142855
Book : 39758.5
Food & Drink : 33333.92307692308
Finance : 31467.944444444445
Photo & Video : 28441.54375
Travel : 28243.8
Shopping : 26919.690476190477
Health & Fitness : 23298.015384615384
Sports : 23008.898550724636
Games : 22788.6696905016
News : 21248.023255813954
Productivity : 21028.410714285714
Utilities : 18684.456790123455
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Business : 7491.117647058823
Education : 7003.983050847458
Catalogs : 4004.0
Medical : 612.0


In [196]:
# Apps Categoria Referencia
for app in ios_free:
    if app[-5] == 'Food & Drink':
        print(app[1], ':', app[5])

Starbucks : 303856
Domino's Pizza USA : 258624
OpenTable - Restaurant Reservations : 113936
Allrecipes Dinner Spinner : 109349
DoorDash - Food Delivery : 25947
UberEATS: Uber for Food Delivery : 17865
Postmates - Food Delivery, Faster : 9519
Dunkin' Donuts - Get Offers, Coupons & Rewards : 9068
Chick-fil-A : 5665
McDonald's : 4050
Deliveroo: Restaurant Delivery - Order Food Nearby : 1702
SONIC Drive-In : 1645
Nowait Guest : 1625
7-Eleven, Inc. : 1356
Outback : 805
Bon Appetit : 750
Starbucks Keyboard : 457
Whataburger : 197
Delish Eatmoji Keyboard : 154
Lieferheld - Delicious food delivery service : 29
Lieferando.de : 29
McDo France : 22
Chefkoch - Rezepte, Kochen, Backen & Kochbuch : 20
Youmiam : 9
Marmiton Twist : 2
Open Food Facts : 1


In [172]:
# Apps Categoria Navigation
for app in ios_free:
    if app[-5] == 'Navigation':
        print(app[1], ':', app[5])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


In [173]:
# Apps Categoria Social Networking
for app in ios_free:
    if app[-5] == 'Social Networking':
        print(app[1], ':', app[5])

Facebook : 2974676
Pinterest : 1061624
Skype for iPhone : 373519
Messenger : 351466
Tumblr : 334293
WhatsApp Messenger : 287589
Kik : 260965
ooVoo – Free Video Call, Text and Voice : 177501
TextNow - Unlimited Text + Calls : 164963
Viber Messenger – Text & Call : 164249
Followers - Social Analytics For Instagram : 112778
MeetMe - Chat and Meet New People : 97072
We Heart It - Fashion, wallpapers, quotes, tattoos : 90414
InsTrack for Instagram - Analytics Plus More : 85535
Tango - Free Video Call, Voice and Chat : 75412
LinkedIn : 71856
Match™ - #1 Dating App. : 60659
Skype for iPad : 60163
POF - Best Dating App for Conversations : 52642
Timehop : 49510
Find My Family, Friends & iPhone - Life360 Locator : 43877
Whisper - Share, Express, Meet : 39819
Hangouts : 36404
LINE PLAY - Your Avatar World : 34677
WeChat : 34584
Badoo - Meet New People, Chat, Socialize. : 34428
Followers + for Instagram - Follower Analytics : 28633
GroupMe : 28260
Marco Polo Video Walkie Talkie : 27662
Miitomo : 2

In [174]:
# Apps Categoria Music
for app in ios_free:
    if app[-5] == 'Music':
        print(app[1], ':', app[5])

Pandora - Music & Radio : 1126879
Spotify Music : 878563
Shazam - Discover music, artists, videos & lyrics : 402925
iHeartRadio – Free Music & Radio Stations : 293228
SoundCloud - Music & Audio : 135744
Magic Piano by Smule : 131695
Smule Sing! : 119316
TuneIn Radio - MLB NBA Audiobooks Podcasts Music : 110420
Amazon Music : 106235
SoundHound Song Search & Music Player : 82602
Sonos Controller : 48905
Bandsintown Concerts : 30845
Karaoke - Sing Karaoke, Unlimited Songs! : 28606
My Mixtapez Music : 26286
Sing Karaoke Songs Unlimited with StarMaker : 26227
Ringtones for iPhone & Ringtone Maker : 25403
Musi - Unlimited Music For YouTube : 25193
AutoRap by Smule : 18202
Spinrilla - Mixtapes For Free : 15053
Napster - Top Music & Radio : 14268
edjing Mix:DJ turntable to remix and scratch music : 13580
Free Music - MP3 Streamer & Playlist Manager Pro : 13443
Free Piano app by Yokee : 13016
Google Play Music : 10118
Certified Mixtapes - Hip Hop Albums & Mixtapes : 9975
TIDAL : 7398
YouTube Mu

In [193]:
# Apps Categoria Book
for app in ios_free:
    if app[-5] == 'Finance':
        print(app[1], ':', app[5])

Chase Mobile℠ : 233270
Mint: Personal Finance, Budget, Bills & Money : 232940
Bank of America - Mobile Banking : 119773
PayPal - Send and request money safely : 119487
Credit Karma: Free Credit Scores, Reports & Alerts : 101679
Capital One Mobile : 56110
Citi Mobile® : 48822
Wells Fargo Mobile : 43064
Chase Mobile : 34322
Square Cash - Send Money for Free : 23775
Capital One for iPad : 21858
Venmo : 21090
USAA Mobile : 19946
TaxCaster – Free tax refund calculator : 17516
Amex Mobile : 11421
TurboTax Tax Return App - File 2016 income taxes : 9635
Bank of America - Mobile Banking for iPad : 7569
Wells Fargo for iPad : 2207
Stash Invest: Investing & Financial Education : 1655
Digit: Save Money Without Thinking About It : 1506
IRS2Go : 1329
Capital One CreditWise - Credit score and report : 1019
U by BB&T : 790
Paribus - Rebates When Prices Drop : 768
KeyBank Mobile : 623
VyStar Mobile Banking for iPhone : 434
Sparkasse - Your mobile branch : 77
VyStar Mobile Banking for iPad : 57
Zaim : 4

## Analisando gêneros com maior download

Como pode ser visto acima, os gêneros com maior download são: 

- Navigation : 86090.3
- Reference : 74942.1
- Social Networking : 71548.35
- Music : 57326.53
- Weather : 52279.89
- Book : 39758.5
- Food & Drink : 33333.92
- Finance : 31467.94
- Photo & Video : 28441.54
- Travel : 28243.8

O gênero Navigation e Social Networking não podem ser recommendados pois são dominados por gigantes no mercado, como Instagram, Facebook, Waze e Maps. Fazendo com que se removidos, a importância do gênero caia drasticamente. 

Os gêneros Music, Travel, Book, Weather, Finance, Photo & Video e Food & Drink, seguem uma dinâmica semelhante.

Já o gênero Reference possui a maioria dos downloads por causa dos apps bíblia e dicionário. Ou seja, livros famosos de refência. 
Então o genre Reference, também não pode ser considerado como um perfil recomendado, á não ser que se crie apps com livros famosos de refência que passem a ter a importancia que um dicionário tem.

Por ser da área de eletrônica, creio ser bastante fácil criar um app de jogo e publicar na loja. Por esse motivo e também analisando os dados da coluna Games, posso recomendar esse perfil, pois na coluna Games muitos jogos possuem um número de downloads gigantescos e não somente alguns como acontece nas outras colunas. Sua média ficou baixa pois o número de jogos da loja é bem grande.


## Analisando apps ANDROID 

A coluna que possui o número de downloads dos apps do android possui caracteres especiais e para computar as médias, precisamos converte-los para *float*.

Após feito isso, vamos verificar quais categorias possuem maior numero de instalações.

In [203]:
dic2 = {}
category = tabela_freq(android_free, 1)

for genre in category:
    total = 0
    len_category = 0
    
    for app in android_free:
        category_app = app[1]
        if genre == category_app:
            n_installs = app[5]
            n_installs = n_installs.replace("+","")
            n_installs = float(n_installs.replace(",",""))
            total += n_installs
            len_category += 1
            
    avg_installs = total/len_category
    dic2[genre] = avg_installs
    #print("Categoria: ", genre, "| Média: ", avg_installs )

In [202]:
table_display=[]
for key in dic2:
    key_val_as_tuple = (dic2[key], key)
    table_display.append(key_val_as_tuple)

table_sorted = sorted(table_display, reverse = True)
for entry in table_sorted:
    print(entry[1], ':', entry[0])

COMMUNICATION : 38456119.167247385
VIDEO_PLAYERS : 24727872.452830188
SOCIAL : 23253652.127118643
PHOTOGRAPHY : 17840110.40229885
PRODUCTIVITY : 16787331.344927534
GAME : 15588015.603248259
TRAVEL_AND_LOCAL : 13984077.710144928
ENTERTAINMENT : 11640705.88235294
TOOLS : 10801391.298666667
NEWS_AND_MAGAZINES : 9549178.467741935
BOOKS_AND_REFERENCE : 8767811.894736841
SHOPPING : 7036877.311557789
PERSONALIZATION : 5201482.6122448975
WEATHER : 5074486.197183099
HEALTH_AND_FITNESS : 4188821.9853479853
MAPS_AND_NAVIGATION : 4056941.7741935486
FAMILY : 3695641.8198090694
SPORTS : 3638640.1428571427
ART_AND_DESIGN : 1986335.0877192982
FOOD_AND_DRINK : 1924897.7363636363
EDUCATION : 1833495.145631068
BUSINESS : 1712290.1474201474
LIFESTYLE : 1437816.2687861272
FINANCE : 1387692.475609756
HOUSE_AND_HOME : 1331540.5616438356
DATING : 854028.8303030303
COMICS : 817657.2727272727
AUTO_AND_VEHICLES : 647317.8170731707
LIBRARIES_AND_DEMO : 638503.734939759
PARENTING : 542603.6206896552
BEAUTY : 51315

In [208]:
# Apps Categoria Book
for app in android_free:
    if app[1] == 'GAME':
        print(app[0], ':', app[5])

Solitaire : 10,000,000+
Sonic Dash : 100,000,000+
PAC-MAN : 100,000,000+
Bubble Witch 3 Saga : 50,000,000+
Race the Traffic Moto : 10,000,000+
Marble - Temple Quest : 10,000,000+
Shooting King : 10,000,000+
Geometry Dash World : 10,000,000+
Jungle Marble Blast : 5,000,000+
Roll the Ball® - slide puzzle : 100,000,000+
Block Craft 3D: Building Simulator Games For Free : 50,000,000+
Farm Fruit Pop: Party Time : 1,000,000+
Love Balls : 50,000,000+
Piano Tiles 2™ : 100,000,000+
Pokémon GO : 100,000,000+
Paint Hit : 10,000,000+
Snake VS Block : 50,000,000+
Rolly Vortex : 10,000,000+
Woody Puzzle : 1,000,000+
Stack Jump : 10,000,000+
The Cube : 5,000,000+
Extreme Car Driving Simulator : 100,000,000+
Bricks n Balls : 1,000,000+
The Fish Master! : 1,000,000+
Color Road : 10,000,000+
Draw In : 10,000,000+
PLANK! : 500,000+
Looper! : 1,000,000+
Trivia Crack : 100,000,000+
Will it Crush? : 5,000,000+
Tomb of the Mask : 5,000,000+
Baseball Boy! : 10,000,000+
Hello Stars : 10,000,000+
Tank Stars : 1

Analisando as categorias dos apps mais baixados para android, observa-se o mesmo inviesamento dos apps para ios. As gigantes do mercado lideram disparado e por isso não podemos recomendar esses generos como perfil apropriado 

# Conclusão

Apesar do gênero Games estar um tanto saturado, em ambas as lojas esse gênero ainda possui um alto número de downloads distribuidos em um número alto de apps.
Assim, creio que esse gênero seja uma boa opção de perfil para novos apps que gerem bastante número de usuários.